src/librustc/session/config.rs

```diff
@@ -1095,6 +1133,16 @@ macro_rules! options {
             true
         }
 
+        fn parse_tracer(slot: &mut TracerMode, v: Option<&str>) -> bool {
+            *slot = match v {
+                None | Some("off") => TracerMode::Off,
+                Some("sw") => TracerMode::Software,
+                Some("hw") => TracerMode::Hardware,
+                Some(_) => return false,
```

Comment:
Should we be `panic`ing if we get passed something we don't know how to deal with?

Reply:
Returning false tells the compiler to give an appropriate error message:
```
$ rustc -C tracer=taps src/main.rs 
error: incorrect value `taps` for codegen option `tracer` - one of `off`, `sw`, `sw-nosir` or `hw` was expected
```

But we do have a bug in the message! `sw-nosir` is gone! Fixing.

Reply:
OK.

Reply:
Fixed